contrib/platform/src/com/sun/jna/platform/win32/Winspool.java

```diff
@@ -499,7 +499,97 @@ public PRINTER_INFO_4(int size) {
             return FIELDS;
         }
     }
+	
+	/**
+	 * The FORM_INFO_2 structure specifies general printer information.
+	 *
+	 * @see <a href=
+	 *      "https://msdn.microsoft.com/en-us/library/windows/desktop/dd162844(v=vs.85).aspx">
+	 *      FORM_INFO_2 structure</a>
+	 */
+	public static class FORM_INFO_2 extends Structure {
+		
+		public static final List<String> FIELDS = createFieldsOrder("Flags", "pName", "Size", "ImageableArea", "pKeyword", "StringType", "pMuiDll", "dwResourceId", "pDisplayName", "wLangId");
+		
+		/**
+		 * Specifies information about the returned data. Following are the
+		 * values for this member.
+		 */
+		public int Flags;
+		/**
+		 * Pointer to a null-terminated string that names the contents of the
+		 * structure.
+		 */
+		public String pName;
+		
+		public SIZEL Size;
+		
+		public RECTL ImageableArea;
+		
+		public String pKeyword;
```

Comment:
This is problematic. Winspool is loaded with W32APIOptions.DEFAULT_OPTIONS, so the value of the system property "w32.ascii" determines whether the unicode or the ascii variant of the functions are called. The structures are mapped following this by default. You can see this in the structure definitions/return types/parameters:

LPTSTR is a sure sign. An LPTSTR is depending on the context either an LPSTR (ascii variant) or an LPWSTR (unicode variant). (See: https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751(v=vs.85).aspx#LPTSTR)

This structure is problematic as here the strings are mapped inconsistently. pKeyword is a LPCSTR, while all other strings are LPTSTR. So you'll need a special typemapper, that takes the StructureReadContext and either delegegates to the com.sun.jna.win32.W32APITypeMapper#DEFAULT (for normal fields) and special cases pKeyword. (No I did not work with that yet, but sounds interesting :-))